packages/sitecore-jss/src/graphql-request-client.test.ts

```diff
@@ -330,6 +387,76 @@ describe('GraphQLRequestClient', () => {
         });
       }
 
+      const retryableErrorCodeThrowError = async (errorCode: string) => {
+        nock('http://jssnextweb')
+          .post('/graphql')
+          .replyWithError({ code: errorCode })
+          .post('/graphql')
+          .replyWithError({ code: errorCode })
+          .post('/graphql')
+          .replyWithError({ code: errorCode });
+
+        const graphQLClient = new GraphQLRequestClient(endpoint, {
+          retries: 2,
+        });
+
+        spy.on(graphQLClient['client'], 'request');
+
+        try {
+          await graphQLClient.request('test');
+        } catch (error) {
+          expect(error).to.not.be.undefined;
+          expect(graphQLClient['client'].request).to.have.been.called.exactly(3);
+          spy.restore(graphQLClient);
+        }
+      };
+
+      // Test cases for each retryable error codes
+      for (const errorCode of ['ECONNRESET', 'ETIMEDOUT', 'EPROTO']) {
+        it(`should retry and throw error for ${errorCode} when retries specified`, async function() {
+          this.timeout(8000);
+          await retryableErrorCodeThrowError(errorCode);
+        });
+      }
+
+      const retryableErrorCodeResolve = async (errorCode: string) => {
+        nock('http://jssnextweb')
```

Comment:
I don't think we can refactor them into a separate function since the response for node error codes is different and for client error codes is different. Also, for node error codes we use `replyWithError` and for client error codes we use reply for response in `nock`. This would cause an issue for creating a common nock request response.